test/integration/h2_fuzz.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+#include "test/common/http/http2/http2_frame.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+#include "test/integration/h2_capture_fuzz.pb.h"
+#include "test/integration/http_integration.h"
+
+namespace Envoy {
+
+class H2FuzzIntegrationTest : public HttpIntegrationTest {
+public:
+  H2FuzzIntegrationTest(Network::Address::IpVersion version)
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP2, version) {}
+
+  void initialize() override;
+  void replay(const test::integration::H2CaptureFuzzTestCase&);
+
+private:
+  // Maps between the enum value in h2_capture_fuzz.proto to the correct
+  // function in test/common/http/http2/http2_frame.h
+  // TODO(adip): some methods receive the stream index, which is currently set to 1, this should
+  // be fuzzed further
```

Comment:
Yeah, I'd suggest making the data model a `oneof` instead of `enum`. That way, each frame will have a corresponding `message` where we can place the stream index.

Reply:
Modified. Also added other sub-fields, that can have different values.
For now, it's all based on http2_frame methods, but can be extended if needed.